hw/ip/prim/README.md

```diff
@@ -4,96 +4,156 @@
+Tools that perform this lowering undertake a mapping process where higher-level functionality is decomposed into lower and lower level components which exist in or can be created in the physical hardware or technology that is being targeted.
+In ASIC or FPGA targets, these lower level components may be macro cells or other fixed-function hardware blocks.
+An alternative lowering could be to gate-level models of an equivalent hardware implementation, where additional information is added to increase simulation accuracy for prediction of timing and power characteristics.
+These tools are described as 'inferring' a combination of lower level components that best implement the circut functionality that is described.
+There are many reasons that we may wish to constrain or modify our source files to control inferencing of a set of hardware components to make up a final circuit.
+
+## Primitive Implementations & Libraries
+
+Primitives in OpenTitan exist at a low RTL abstraction level where it may be desirable to swap-out implementations to obtain better optimized hardware under different circumstances.
+For this reason, a Primitive is defined as a module with a fixed name and set of ports, to allow implementations to be easily interchanged.
+
+However, not all primitives need to have multiple implementations.
+
+- Primitives with a single generic implementation, where any lowering process is able to infer a suitable hardware implementation from language-level RTL features, are called '***Technology-Independent Primitives***'.
+- Primitives with multiple implementations, each of which may be optimized for or targeted towards a certain hardware platform, are called '***Technology-Dependent Primitives***'.
 
-Under the hood, primitives are slightly special, as they can have multiple implementations.
-In contrast to many other modules in a hardware design, primitives must often be implemented in technology-dependent ways.
-For example, a clock multiplexer for a Xilinx FPGA is implemented differently than one for a specific ASIC technology.
+Technology-dependent primitive implementations may be grouped together into '***Technology Libraries***'.
+For example, a technology library of primitive implementations may be created for a certain ASIC technology, where the implementations are optimized for desirable synthesis characteristics in that technology.
+Within the OpenTitan repository, each technology library has its own directory within `hw/ip/` with the prefix `prim`, such as `hw/ip/prim_xilinx/` for synthesis targeting Xilinx FPGAs.
 
-Not all primitives need to have multiple implementations.
+For each technology-dependent primitive, there should be an implementation that is 'generic' in that it is constructed using language-level RTL constructs, and can be consumed by most downstream tooling.
+These implementations are commonly used as input to simulation engines for verification and as a functional reference.
 
```

Comment:
I'd drop this gap (which makes things more symmetrical with the previous paragraph?)